src/plugins/stylesheet/CSSResourcePlugin.ts

```diff
@@ -190,16 +203,54 @@ export class CSSResourcePluginClass implements Plugin {
             if (this.distFolder) {
                 let newFileName = this.useOriginalFilenames ? path.relative(file.context.homeDir, urlFile) : generateNewFileName(urlFile);
 
-                if (!files[urlFile]) {
+                if (!this.files[urlFile]) {
                     let newPath = path.join(this.distFolder, newFileName);
                     tasks.push(copyFile(urlFile, newPath));
-                    files[urlFile] = true;
+                    this.files[urlFile] = true;
+                    
+                    // We store this copied file source and destination path for the middleware
+                    this.copiedFiles.push({
+                        from: urlFile,
+                        to: newPath
+                    });
+
+                    // We also store a string which uniquely identify this array
+                    // To avoid watch loop
+                    this.copiedFilesID = this.copiedFiles.map(
+                        copiedFile => generateNewFileName( copiedFile.from )
+                    ).join('+');
                 }
                 return this.resolveFn(newFileName);
             }
         }
         file.contents = CSSUrlParser.walk(file.contents, walker);
     }
+    
+
+    bundleEnd (producer)
+    {
+        // If there is no middleware from config, quit
+        if (!this.filesMapping) return;
+
+        // Get home dir (src path)
+        const homeDir = producer.fuse.opts.homeDir;
+
+        // We store the copied file ID and continue only if it changed
+        // It allow us to avoid watch loop when fileMapping middleware is producing a source file
+        if (this.previousCopiedFilesID === this.copiedFilesID) return;
+        this.previousCopiedFilesID = this.copiedFilesID;
+
+        // Call middleware with copied files path
+        this.filesMapping(
```

Comment:
How about if filesMapping isn't defined, will it be still copied over?

Reply:
We are checking if this.fileMapping is defined from options at line 233. If it is not defined from options, file are still copied. Maybe this isn't clear enough ?

Reply:
Right, I just though that you changed the copying logic too. Can you update the doc too?

Reply:
I'll try to do it but it will be in an another pull request I think.